### PR TITLE
cli: show download progress,  add custom mirror

### PR DESCRIPTION
**Pull request by FeelyChau**
Show download progress when download framework and scripts.
Because the function `downloadAndExtractTo` with progress should not used by scripts (not all the runtime run in terminal), I copy and modify it in `cli`.

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -19,9 +19,11 @@
   "license": "Apache 2.0",
   "dependencies": {
     "@pipcook/boa": "^2.0.0-beta.1",
-    "@pipcook/costa": "^2.0.0-beta.1",
     "@pipcook/core": "^2.0.0-beta.1",
+    "@pipcook/costa": "^2.0.0-beta.1",
+    "@types/cli-progress": "^3.9.1",
```
Comment (yorkie) [inline packages/cli/package.json:24]: Type packages should be in devDependencies.

**Review by yorkie — CHANGES_REQUESTED** [review]

**Comment by rickycao-qy** [comment]
CI is failed

**Review by WenheLI — APPROVED** [review]
LGTM

**Review by yorkie — APPROVED** [review]

**Review by rickycao-qy — APPROVED** [review]
LGTM

**Review by WenheLI — APPROVED** [review]

**Pull request merged**